src/Data/EuclideanRing.purs

```diff
@@ -8,16 +8,27 @@ module Data.EuclideanRing
 import Data.CommutativeRing (class CommutativeRing)
 import Data.Ring (class Ring, sub, (-))
 import Data.Semiring (class Semiring, add, mul, one, zero, (*), (+))
-import Data.Unit (Unit, unit)
 
 -- | The `EuclideanRing` class is for commutative rings that support division.
+-- | The mathematical structure this class is based on is sometimes also called
+-- | a *Euclidean domain*.
 -- |
--- | Instances must satisfy the following law in addition to the `Ring`
+-- | Instances must satisfy the following laws in addition to the `Ring`
 -- | laws:
 -- |
--- | - Integral domain: `a /= 0` and `b /= 0` implies `a * b /= 0`
--- | - Multiplicative Euclidean function: ``a = (a / b) * b + (a `mod` b)``
--- |   where `degree a > 0` and `degree a <= degree (a * b)`
+-- | - Integral domain: `one /= zero`, and if `a` and `b` are both nonzero then
+-- |   so is their product `a * b`.
+-- | - Euclidean function `degree`:
+-- |   - For all `a` and `b`, where `b /= zero`, let `q = a / b` and
+-- |     ``r = a `mod` b``. Then `a = q*b + r`, and also either `r = zero` or
+-- |     `degree r < degree b`.
+-- | - Submultiplicative euclidean function:
+-- |   - For all nonzero `a` and `b`, `degree a <= degree (a * b)`.
+-- |
+-- | For any `EuclideanRing` which is also a `Field`, one valid choice
+-- | for `degree` is simply `const one`. In fact, unless there's a specific
```

Comment:
Minorest of nitpicks, could say `const 1` here since `degree` returns an `Int` rather than an `a`.